sklearn/decomposition/dict_learning.py

```diff
@@ -1081,6 +1307,184 @@ def fit(self, X, y=None):
+
+    random_state : int or RandomState
+        Pseudo number generator state used for random sampling.
+
+    Attributes
+    ----------
+    components_ : array, [n_components, n_features]
+        components extracted from the data
+
+    inner_stats_ : tuple of (A, B) ndarrays
+        Internal sufficient statistics that are kept by the algorithm.
+        Keeping them is useful in online settings, to avoid loosing the
+        history of the evolution, but they shouldn't have any use for the
+        end user.
+        A (n_components, n_components) is the dictionary covariance matrix.
+        B (n_features, n_components) is the data approximation matrix
+
+    n_iter_ : int
+        Number of iterations run.
+
+    Notes
+    -----
+    **References:**
+
+    Olshausen BA, Field DJ (1996). 
+    Emergence of simple-cell receptive field properties by learning a sparse code for natural images. 
+    Nature, 381: 607-609. (http://redwood.berkeley.edu/bruno/papers/nature-paper.pdf)
+
+    Olshausen BA, Field DJ (1997) 
+    Sparse Coding with an Overcomplete Basis Set: A Strategy Employed by V1?  
+    Vision Research, 37: 3311-3325.   (http://redwood.berkeley.edu/bruno/papers/VR.pdf)
+
+    See also
+    --------
+    SparseCoder
+    DictionaryLearning
+    MiniBatchDictionaryLearning
+    SparsePCA
+    MiniBatchSparsePCA
+
+    """
+    def __init__(self, eta=0.02, n_components=None, n_iter=40000,
+                 gain_rate=0.001, alpha_homeo=0.02,
+                 fit_algorithm='omp', n_jobs=1, batch_size=100,
+                 shuffle=True, dict_init=None, transform_algorithm='omp',
+                 transform_n_nonzero_coefs=None, transform_alpha=None,
+                 verbose=False, split_sign=False, random_state=None):
+
+        self._set_sparse_coding_params(n_components, transform_algorithm,
+                                       transform_n_nonzero_coefs,
+                                       transform_alpha, split_sign, n_jobs)
+        self.transform_n_nonzero_coefs = transform_n_nonzero_coefs
+        self.eta = eta
+        self.n_iter = n_iter
+        self.gain_rate = gain_rate
+        self.alpha_homeo = alpha_homeo
+        self.fit_algorithm = fit_algorithm
+        self.dict_init = dict_init
+        self.verbose = verbose
+        self.shuffle = shuffle
+        self.batch_size = batch_size
+        self.split_sign = split_sign
+        self.random_state = random_state
+
+    def fit(self, X, y=None):
+        """Fit the model from data in X.
+
+        Parameters
+        ----------
+        X: array-like, shape (n_samples, n_features)
+            Training vector, where n_samples in the number of samples
+            and n_features is the number of features.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        random_state = check_random_state(self.random_state)
+        X = check_array(X)
```

Comment:
you should perhaps use ensure_min_samples here, as the code is currently failing for a single-sample input.